### PR TITLE
Add micro benchmark framework

### PR DESCRIPTION
**Pull request by fmatthew5876**
https://github.com/google/benchmark

Enabled by `-DPLAYER_ENABLE_BENCHMARKS=ON` in cmake. Turned off by default.

Only when build is enabled, we pull in the libbenchmark dependency.

I used this to get the numbers in #1959

Also adds a bunch of smaller changes to make it easier to write automated tests

### CODE REVIEW

File: src/cache.cpp [hunk 1]
```diff
@@ -55,30 +59,29 @@ namespace {
...
+		auto cur_ticks = clock::now();
 
 		for (auto& i : cache) {
 			if (i.second.bitmap.use_count() != 1) {
 				// Bitmap is referenced
 				continue;
 			}
 
-			if (cache_size <= cache_limit && cur_ticks - i.second.last_access < 3000) {
+			if (cache_size <= cache_limit && cur_ticks - i.second.last_access < 3000s) {
```
Comment (Ghabry) [inline src/cache.cpp]: is this correct? The original check is for 3000ms/3s

**Comment by Ghabry** [comment]
The benchmark stuff is not invasive and disabled by default, so I'm fine with this change.

Only problem I see is that contrary to unit tests the benchmarks are not part of PR reviews, so they will likely break when the API changes.

File: src/cache.cpp (hunk 1 above)
Comment (carstene1ns) [reply src/cache.cpp]: yeah, he force-pushed this away. seems like an oversight.

Comment (fmatthew5876) [reply src/cache.cpp]: Was a bug on my part, fixed.

**Comment by fmatthew5876** [comment]
> Only problem I see is that contrary to unit tests the benchmarks are not part of PR reviews, so they will likely break when the API changes.

We always have the option to enable them in PR builds if we want. Maybe only for a subset if the library dependency becomes onerous.

**Pull request merged**